system/pamd.py

```diff
@@ -0,0 +1,418 @@
+
+        if stringline is not None and (rule_type is not None or
+                                       rule_control is not None or
+                                       rule_module_path is not None):
+            ansible.fail_json(msg='This is bizarre.  You can only pass' +
+                              'in a line OR a type, control, and module path')
+        elif stringline is not None:
+            self.line = stringline
+            self.__parsline__()
+        else:
+            self.rule_type = rule_type
+            self.rule_control = rule_control
+            self.rule_module_path = rule_module_path
+            
+            try:
+                if self.rule_module_args is not None and type(self.rule_module_args) is list:
+                    self.rule_module_args = rule_module_args
+                elif self.rule_module_args is not None and type(self.rule_module_args) is str:
+                    self.rule_module_args = rule_module_args.split(' ')
+            except AttributeError:
+                self.rule_module_args = []
+
+    def __parsline__(self):
+
+        split_line = self.line.split()
+        self.rule_type = split_line[0]
+        self.rule_control = split_line[1]
+
+        if self.rule_control.startswith('['):
+            self.rule_control = self.line[self.line.index('['):self.line.index(']')+1]
+
+        if "]" in split_line[2]:
+            self.rule_module_path = split_line[3]
+            self.rule_module_args = split_line[4:]
+        else:
+            self.rule_module_path = split_line[2]
+            self.rule_module_args = split_line[3:]
+
+    def get_module_args_as_string(self):
+        try:
+            if self.rule_module_args is not None:
+                return ' '.join(self.rule_module_args)
+        except AttributeError:
+            pass
+        return ''
+
+    def __str__(self):
+        return "%-10s\t%s\t%s %s" % (self.rule_type, self.rule_control, self.rule_module_path, self.get_module_args_as_string())
+
+#PamdService encapsulates an entire service and contains one or more rules
+class PamdService(object):
+
+    def __init__(self, path, name, ansible):
+        self.path = path
+        self.name = name
+        self.check = ansible.check_mode
+        self.ansible = ansible
+        self.fname = self.path + "/" + self.name
+        self.preamble = []
+        self.rules = []
+
+        if os.path.exists(self.fname):
+            for line in fileinput.FileInput(self.fname, inplace=False):
+                if line.startswith('#') and not line.isspace():
+                    self.preamble.append(line.rstrip())
+                elif not line.startswith('#') and not line.isspace():
+                    self.rules.append(PamdRule(ansible, stringline=line.rstrip()))
+            fileinput.close()
+        else:
+            self.ansible.fail_json(msg='Unable to locate PAM module file %s' % self.fname)
+    
+    def __str__(self):
+        return self.fname
+
+class PamdUtil(object):
```

Comment:
Since this is all static methods, making them module scope methods would be more traditional.
But the class is fine as well.